### PR TITLE
GPII-3811: Make custom high-contrast themes work with updated Windows.

### PR DESCRIPTION
**Pull request by stegru**
Still unsure of the underlying cause, but here's a workaround.

After the theme has been applied, this fix just reads the .theme file and manually applies the colours.

Works well with https://github.com/GPII/universal/pull/761

### CODE REVIEW

**Comment by stegru** [comment]
ok to test

```
    default: /vagrant => /home/jenkins-gpii/workspace/windows-tests
Vagrant can't use the requested machine because it is locked! This
means that another Vagrant process is currently reading or modifying
the machine. Please wait for that Vagrant process to end and try
again. Details about the machine are shown below:

Name: default
Provider: virtualbox
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

File: gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js [hunk 1]
```diff
@@ -619,6 +623,75 @@ gpii.windows.spiSettingsHandler.saveTheme = function (currentThemeFile, saveAs)
...
+ * @param {Boolean} allThemes true to apply every theme, rather than just the custom ones.
+ */
+gpii.windows.spiSettingsHandler.applyCustomTheme = function (themeFile, allThemes) {
+    var themeData = gpii.iniFile.readFile(themeFile);
+    fluid.log("applyCustomTHeme", arguments);
+
+    var isValid = themeData && themeData.Theme;
+    // The custom themes, unlike the built-in ones, have the ThemeId value set.
+    if (isValid && (themeData.Theme.ThemeId || allThemes)) {
+        // The list of colour names, whose index is the value of the corresponding API constant.
```
Comment (amb26) [inline gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js]: The API constant held where? Can you supply a link - also, please hoist this into a global constant

File: gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js [hunk 2]
```diff
@@ -619,6 +623,75 @@ gpii.windows.spiSettingsHandler.saveTheme = function (currentThemeFile, saveAs)
...
+        var colorValues = [];
+        fluid.each(colors, function (rgbValues, name) {
+            colorNumbers.push(colorNames.indexOf(name));
+            var rgb = rgbValues.split(" ");
+            colorValues.push(rgb[0] | rgb[1] << 8 | rgb[2] << 16);
+        });
+
+
+        /*eslint-env es6, Int32Array is undefined */
+        var elementsBuffer = Int32Array.from(colorNumbers);
```
Comment (amb26) [inline gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js]: Safer to stuff Int32Array into eslint globals than open the floodgates to arbitrary ES6 junk : P

File: gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js [hunk 3]
```diff
@@ -514,7 +514,7 @@ gpii.windows.spi.GPII1873_HighContrastBug = function (method, payload, results)
  * @param {String} newThemeFile The .theme file, which is in INI file format.
  * @param {String} currentThemeFile The current .theme file, which is in INI file format.
  * @param {String} saveAs The .theme file to which the current theme is saved.
- * @param {Boolean} dryRun true to not update the registry.
+ * @param {Boolean}  dryRun true to not update the registry.
```
Comment (javihernandez) [inline gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js]: why an extra space?

File: gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js (hunk 1 above)
Comment (stegru) [reply gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js]: done

File: gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js (hunk 2 above)
Comment (stegru) [reply gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js]: yes, done

File: gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js (hunk 3 above)
Comment (stegru) [reply gpii/node_modules/spiSettingsHandler/src/SpiSettingsHandler.js]: fixed

**Pull request merged**